File: Java/Easy/Remove Nth Node From End of List.java
public class Solution {
    /**
     * @param head: The first node of linked list.
     * @param n: An integer.
     * @return: The head of linked list.
     */
    ListNode removeNthFromEnd(ListNode head, int n) {
        // write your code here
        int count = 1;
        ListNode curt = head;
        if(curt.next == null) {
            return null;
        }
        while(curt!=null) {
            curt = curt.next;
            count++;
        }
        int index = count - n;
        if(index ==1) {
            head = head.next;
            return head;
        }
        curt = head;
        for(int i=1; i<index-1; i++) {
            curt = curt.next;
        }
        curt.next = curt.next.next;
        return head;
    }
}

File: C++/Easy/Remove Nth Node From End of List.cpp
/**
 * Definition of ListNode
 * class ListNode {
 * public:
 *     int val;
 *     ListNode *next;
 *     ListNode(int val) {
 *         this->val = val;
 *         this->next = NULL;
 *     }
 * }
 */
class Solution {
public:
    /**
     * @param head: The first node of linked list.
     * @param n: An integer.
     * @return: The head of linked list.
     */
    ListNode *removeNthFromEnd(ListNode *head, int n) {
        // write your code here
        if(n == 0) {
            return head;
        }
        ListNode *fast = head;
        ListNode *slow = new ListNode(0);
        ListNode *ret = slow;
        slow->next = head;
        for(int i=0; i<n; i++) {
            fast = fast->next;
        }
        while(fast!=NULL) {
            fast = fast->next;
            slow = slow->next;
        }
        slow->next = slow->next->next;
        return ret->next;
    }
};
